source/docs/yearly-changes/yearly-changelog.rst

```diff
@@ -0,0 +1,163 @@
+
+.. code-block:: java
+
+   /* Setting up bindings for necessary control of the swerve drive platform */
+   CommandXboxController joystick = new CommandXboxController(0); // My joystick
+   CommandSwerveDrivetrain drivetrain = TunerConstants.DriveTrain; // My drivetrain
+   SwerveRequest.FieldCentric drive = new SwerveRequest.FieldCentric().withIsOpenLoop(true); // I want field-centric
+                                                                                             // driving in open loop
+   SwerveRequest.SwerveDriveBrake brake = new SwerveRequest.SwerveDriveBrake();
+   SwerveRequest.PointWheelsAt point = new SwerveRequest.PointWheelsAt();
+   Telemetry logger = new Telemetry(MaxSpeed);
+
+   private void configureBindings() {
+      drivetrain.setDefaultCommand( // Drivetrain will execute this command periodically
+      drivetrain.applyRequest(() ->
+         drive.withVelocityX(-joystick.getLeftY() * MaxSpeed) // Drive forward with negative Y (forward)
+         .withVelocityY(-joystick.getLeftX() * MaxSpeed) // Drive left with negative X (left)
+         .withRotationalRate(-joystick.getRightX() * MaxAngularRate) // Drive counterclockwise with negative X (left)
+         )
+      );
+
+      joystick.a().whileTrue(drivetrain.applyRequest(() -> brake));
+      joystick.b().whileTrue(drivetrain
+         .applyRequest(() -> point.withModuleDirection(new Rotation2d(-joystick.getLeftY(), -joystick.getLeftX()))));
+
+      if (Utils.isSimulation()) {
+         drivetrain.seedFieldRelative(new Pose2d(new Translation2d(), Rotation2d.fromDegrees(90)));
+      }
+
+      drivetrain.registerTelemetry(logger::telemeterize);
+   }
+
+.. image:: images/swerve-simulation-video.gif
+   :alt: GIF showing swerve simulation support
+
+.. important:: Swerve API requires all necessary swerve devices to be v6 devices. e.g. 4 drive TalonFX, 4 steer TalonFX, 1 Pigeon 2.0, 4 CANcoders.
+
+Signal Logging
+^^^^^^^^^^^^^^
+
+We've added a comprehensive signal logger API (Java, C++, Python) that represents a real-time capture of signals for supported devices. Signal logging can be useful for analysis of signals over a period of time. In applications, they can be useful for tuning PID gains, characterization of systems, analyzing latency on a system and much more.
+
+Information on configuring the logger and extracting the logs are available in the API Reference and Tuner sections.
+
+.. image:: images/tuner-x-log-extractor.png
+   :alt: Log extractor page in Tuner X
+
+Signal API Improvements
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Users can now disable signals by setting their update rate to 0Hz. To reduce boilerplate when disabling signals, we have added an ``optimizeBusUtilization`` function that takes a list of signals as a parameter. This would automatically disable all signals that are not used.
```

Comment:
optimizeBusUtilization is actually a function on the device, and we have optimizeBusUtilizationForAll which takes a list of devices to optimize.